go/enclave/components/rollup_consumer.go

```diff
@@ -71,19 +69,14 @@ func (rc *rollupConsumerImpl) ProcessL1Block(b *common.BlockAndReceipts) (*commo
 		return nil, nil //nolint:nilnil
 	}
 
-	rollup, err := rc.getCanonicalRollup(rollups, b)
-	if err != nil {
-		return nil, err
-	}
-
-	err = rc.processRollup(b.Block, rollup)
+	rollup, err := rc.getSignedRollup(rollups, b)
```

Comment:
I think you need to get all signed rollups and process them one by one as if they came independently.